### PR TITLE
fix: Enhance URL handling in table by rendering URL column types with `<a>` tag.

### LINKED ISSUE #24769: [Bug]: Unable to right click on the link present in URL column type of table widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When a column is of type URL, it is observed that we cannot right click on the link.

### Steps To Reproduce

1. Dnd table and bind a data to it such that it has a URL column type.
2. Observe that you cannot right click on the links present in this column type

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
<ins>Problem</ins>

URLs in table were not being rendered as links, resulting in inconsistent user experience(missing context menus.

<ins>Root cause</ins>

URLs were rendered in `<div>` instead of `<a>`, making the component lack links related features..

<ins>Solution</ins>

This PR handles... 

- Rendering URLs as links in BasicCell for a better user experience.
- Adding specific types for column properties for more robust data validation and type checking.
- Adding unit tests for BasicCell functionality to ensure accurate rendering and behavior.

- Simplifies the AutoToolTipComponent by removing unncessary `LinkWrapper` component

Fixes #24769
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Table"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11681339029>
> Commit: b7c5d176b35407923a120bb19e40252e3a61b628
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11681339029&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Table`
> Spec:
> <hr>Tue, 05 Nov 2024 10:23:38 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Release Notes

- **New Features**
	- Enhanced type safety for `compactMode` and `columnType` properties across various components.
	- Improved rendering logic in the `AutoToolTipComponent` for better control based on `columnType`.
	- Optimized rendering in the `BasicCell` component using `useMemo`.

- **Bug Fixes**
	- Resolved inconsistencies in type definitions for `BasicCell`, `PlainTextCell`, and `SelectCell` components.
	- Updated tooltip behavior in the `AutoToolTipComponent` to ensure accurate rendering.

- **Tests**
	- Introduced a new test suite for the `BasicCell` component, ensuring proper rendering and interaction behaviors.
	- Refined test cases for the `AutoToolTipComponent` to verify accurate rendering under various conditions.
	- Updated test case for URL column verification to check attributes directly instead of navigation.

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/BasicCell.tsx [hunk 1]
```diff
@@ -128,6 +129,14 @@ export const BasicCell = React.forwardRef(
...
       [onEdit, disabledEditIcon, isCellEditable],
     );
+    const contentToRender = useMemo(() => {
+      switch (columnType) {
+        case ColumnTypes.URL:
+          return <a href={url}>{value}</a>;
+        default:
+          return value;
+      }
+    }, [columnType, url, value]);
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/widgets/TableWidgetV2/component/cellComponents/BasicCell.tsx]: Done.

**Review by jsartisan — DISMISSED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**